### PR TITLE
Support `Home` and `End` keys in tabs

### LINKED ISSUE #38471: Improve tab navigations accessibility with `Home` and `End` keys
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed feature requests
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Proposal

I suggest to improve tab navigations (e.g. `.nav-tabs`) accessibility with the keyboard:
- `Home`: moves focus to the **FIRST** tab and activates it.
- `End`: moves focus to the **LAST** tab and activates it.

The idea is taken from the [W3C example](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/examples/tabs-automatic/#kbd_label).

### Motivation and context

This could be useful for users using keyboards, and this enhancement would be the continuation of https://github.com/twbs/bootstrap/pull/33079 (keyboard accessibility with arrow keys).

I've already implemented it in one of my website. This is much the same as the [implementation for arrow keys](https://github.com/twbs/bootstrap/blob/bb33b4bb8fe0d9ce52e6e2753a514700ce58f1bc/js/src/tab.js#L153-L167). My jQuery code looks like below:

```js
$(document).on('keydown', '.nav-link[data-bs-toggle="tab"]', (event) =>
{
	if (['Home', 'End'].includes(event.key))
	{
		event.preventDefault();
		const $tabs = $(event.currentTarget).parents('.nav[role="tablist"]').find('.nav-link[data-bs-toggle="tab"]').not('.disabled');
		const $tabToShow = (event.key === 'Home') ? $tabs.first() : $tabs.last();
		$tabToShow.get(0).focus({
			preventScroll: true
		});

		$tabToShow.tab('show');
	}
});
```

### PR DESCRIPTION
**Pull request by kyletsang**
### Description

Adds support for `Home` and `End` keys in JS tabs.

### Motivation & Context

Better tab accessibility support

### Type of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would change existing functionality)

### Checklist

- [x] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)
- [x] My code follows the code style of the project _(using `npm run lint`)_
- [x] My change introduces changes to the documentation
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

#### Live previews

- <https://deploy-preview-38498--twbs-bootstrap.netlify.app/docs/5.3/components/navs-tabs/#javascript-behavior>

### Related issues

Closes #38471

### CODE REVIEW

**Comment by GeoSot** [comment]
@patrickhlauke is it a navigation behavior we should support or not?

**Comment by patrickhlauke** [comment]
@GeoSot it's not essential, but a nice-to-have https://www.w3.org/WAI/ARIA/apg/patterns/tabs/#keyboardinteraction

File: js/tests/unit/tab.spec.js [hunk 1]
```diff
@@ -630,6 +630,108 @@ describe('Tab', () => {
       expect(spyPrevent).toHaveBeenCalledTimes(2)
     })
 
+    it('if keydown event is Home, handle it', () => {
```
Comment (GeoSot) [inline js/tests/unit/tab.spec.js:633]: Wouldn't be a bit simpler to 'show' programmatically the second link, and after this , trigger this HOME btn, checking a spy or the 'show' event ?

Comment (kyletsang) [reply js/tests/unit/tab.spec.js:633]: I modeled the test to be similar to the existing ones.  I can change this so it's a bit simpler

Comment (kyletsang) [reply js/tests/unit/tab.spec.js:633]: Done @GeoSot

**Comment by mdo** [comment]
@GeoSot @XhmikosR Is this good to go?

File: site/content/docs/5.3/components/navs-tabs.md [hunk 2]
```diff
@@ -567,7 +567,7 @@ And with vertical pills. Ideally, for vertical tabs, you should also add `aria-o
...
+In line with the ARIA Authoring Practices pattern, only the currently active tab receives keyboard focus. When the JavaScript plugin is initialized, it will set `tabindex="-1"` on all inactive tab controls. Once the currently active tab has focus, the cursor keys activate the previous/next tab and the <kbd>Home</kbd> and <kbd>End</kbd> keys activate the first and last tabs respectively. The plugin will change the [roving `tabindex`](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/) accordingly. However, note that the JavaScript plugin does not distinguish between horizontal and vertical tab lists when it comes to cursor key interactions: regardless of the tab list's orientation, both the up *and* left cursor go to the previous tab, and down *and* right cursor go to the next tab.
```
Comment (patrickhlauke) [inline site/content/docs/5.3/components/navs-tabs.md:570]: i'd add a comma just before the `and the <kbd>Home`. so `, and the <kbd>Home`

**Review by GeoSot — APPROVED** [review]

**Comment by GeoSot** [comment]
Seems fine to me 👍

**Review by mdo — APPROVED** [review]

**Pull request merged**